### DISCUSSION TITLE
hide_key not works

### DISCUSSION THREAD

**Discussion by sycolon** [lnoor/sphinx-jsonschema#83] [Q&A]
i have this JSON schema file
```
{
    "$schema": "https://json-schema.org/draft-07/schema#",
    "$id": "https://mywebsite.net/common/message.schema.json",
    "title": "Message",
    "description": "Schema for a message",
    "anyOf": [
        {
            "$ref": "file:definitions.schema.json"
        }
    ]
}
```

i add it in the rst file like this 
`.. jsonschema:: ctrl/common/message.schema.json`

now i want to hide the key $id from presenting in the final page cuz it shows as a link,
i tried in conf.py to add this
```
jsonschema_options = {
    'hide_key': ['$id']
}
```
or this
```
jsonschema_options = {
    'hide_key': '$id'
}
```

or in RST file this :
```
.. jsonschema:: ctrl/common/message.schema.json
   :hide_key: $id
```
i tried to compile with the versions , 1.18.0 / 1.17.2/ 1.17.0

if there any solution i will be thankfull

**Comment by lnoor** [comment]
Hi, I'm afraid you misunderstood the function of hide_key.
Perhaps the documentation is not sufficiently clear.

The purpose of hide_key is to prevent displaying object attributes.
So within
```json
{
    "type": "object",
    "properties": {
        "key1": "value1",
        "key2", "value2"
    }
}
```
You can use hide_key to prevent key1 from being displayed.
The more general use you are looking for is not supported.

I currently have not much time to work on sphinx-jsonschema but I'm willing to consider a PR.
You could extend the function of `:hide_key:` to include metadata as well, or simply render the link as a string instead of a clickable link. Depending on your needs and the amount of time at your disposal (the latter is less work).
